src/loan/protocol_tests.rs

```diff
@@ -171,6 +173,15 @@ mod tests {
             .await
             .unwrap();
 
+        elements_consensus::verify(
+            borrower.collateral_contract().raw_script.clone(),
```

Comment:
I don't think this is quite right. This should be the script is also in the address of the contract, i.e. you need `collateral_contrat().address().script_pubkey`.

Reply:
The entire script of the contract gets put on the witness stack which is inside the serialized transaction (4th parameter).

Reply:
Yes! Good catch! The difference between script and script pubkey still trips me up to this day lol.

Do you think it might be worth being more clear about this in the API? The data type is the same for both script and script pubkey which always makes it easy to mix them up.

Reply:
That would really have to be fixed upstream I think :)

For now, I wanted to keep the API surface simple. What might be useful to build on top of this is to actually take a `TxOut` and check if a certain transaction spends it correctly. But that would require filling in other data (in case you are not getting the UTXO off the chain) which is kind of cumber-some again.